tensorflow_datasets/scripts/documentation/templates/dataset_doc_template.py

```diff
@@ -0,0 +1,382 @@
+  return builder.info.citation
+
+
+def get_figure(builder):
+  if _VISU_DOC_UTIL.has_visualization(builder):
+    return builder.info.full_name
+  return None  # Fuse the sections together if no configs are available
+
+
+# --------------------------- Single builder ---------------------------
+
+def display_builder(builder, sections):
+  return ('\n').join([section.make(builder)
+                      for section in sections if section.make(builder)])
+
+
+# --------------------------- Builder configs ---------------------------
+
+def display_all_builders(builders, all_sections):
+  # For each fields, extract if the field is shared or unique accross builder.
+  common_sections = []
+  unique_sections = []
+  for section in all_sections:
+    if len(set(section.get_signature(b) for b in builders)) == 1:
+      common_sections.append(section)
+    else:
+      unique_sections.append(section)
+
+  common_builder_str = display_builder(next(iter(builders)), common_sections)
+
+  unique_builder_str = []
+  for i, builder in enumerate(builders):
+    header_suffix = ' (default config)' if i == 0 else ''
+    nightly_str = (' ' + _NIGHTLY_DOC_UTIL.icon) \
+        if _NIGHTLY_DOC_UTIL.is_config_nightly(builder) else ''
+    unique_builder_str.append(
+        f'## {builder.name}/{builder.builder_config.name}'
+        f'{header_suffix}{nightly_str}\n')
+    unique_builder_str.append(display_builder(builder, unique_sections))
+  unique_builder_str = ('\n').join(unique_builder_str)
+
+  return common_builder_str + '\n' + unique_builder_str
+
+
+# --------------------------- Main page ---------------------------
+
+def display_builder_configs(builder, config_builders, all_sections):
+  # First case: Single builder
+  if not builder.builder_config:
+    return display_builder(builder, all_sections)
+  # Second case: Builder configs
+  return display_all_builders(config_builders, all_sections)
+
+
+def display_nightly_str(builder):
+  if _NIGHTLY_DOC_UTIL.is_builder_nightly(builder):
+    return f"""\
+Note: This dataset was added recently and is only available in our
+`tfds-nightly` package  {_NIGHTLY_DOC_UTIL.icon}.
+"""
+  if _NIGHTLY_DOC_UTIL.has_nightly(builder):
+    return f"""\
+Note: This dataset has been updated since the last stable release. The new
+versions and config marked with {_NIGHTLY_DOC_UTIL.icon} are only available
+in the `tfds-nightly` package.
+"""
+  return ""
+
+
+def display_manual_instructions(builder):
+  if builder.MANUAL_DOWNLOAD_INSTRUCTIONS:
+    return "Warning: Manual download required. See instructions below."
+  return ""
+
+
+def display_dataset_heading(builder):
+  return f"""\
+# `{builder.name}`
+"""
+
+
+def get_markdown_string(
+    builder,
+    config_builders,
+    visu_doc_util,
+    nightly_doc_util,
+):
+  global _NIGHTLY_DOC_UTIL, _VISU_DOC_UTIL
+
+  _NIGHTLY_DOC_UTIL = nightly_doc_util
+  _VISU_DOC_UTIL = visu_doc_util
+
+  Section = collections.namedtuple('Section', 'get_signature, make')
```

Comment:
Now that doc script is Python 3, using `dataclasses` would be cleaner. And declaration should be moved outside of the function.